### PR TITLE
e2e: 22366-change-language

### PR DESCRIPTION
**Pull request by hjetpoluru**
## **Description**

This is an automation test to validate the change language functionality. 

## **Related issues**
#22366

Fixes:

## **Manual testing steps**

1. Go to User settings page.
2. In the general tab, user can select the language drop and pick any language
3. Selected language should be changed.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by VersoriumX** [comment]
[Update LavaMoat policies](https://github.com/MetaMask/metamask-extension/actions/workflows/update-lavamoat-policies.yml)

File: test/e2e/tests/change-language.spec.js [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+        await driver.delay(2000);
+
+        const labelText = await driver.findElement(
+          By.css('.settings-page__content-label'),
+        );
+        assert.equal(
+          await labelText.getText(),
+          'Current Language',
+          'Language did not change',
+        );
```
Comment (HowardBraham) [inline test/e2e/tests/change-language.spec.js:105]: This is an example of the "find then assert" race condition
https://consensys.slack.com/archives/GTQAGKY5V/p1686238528522549

File: test/e2e/tests/change-language.spec.js [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        // Confirm that the language change is reflected in button
+        const buttonText = await driver.findElement(
+          '[data-testid="auto-lockout-button"]',
+        );
+
+        assert.equal(
+          await buttonText.getText(),
+          'Gem',
+          'Language change is not reflected in button',
+        );
```
Comment (HowardBraham) [inline test/e2e/tests/change-language.spec.js:197]: This is an example of the "find then assert" race condition
https://consensys.slack.com/archives/GTQAGKY5V/p1686238528522549

Comment (hjetpoluru) [reply test/e2e/tests/change-language.spec.js]: I made the code changes and marking resolved.

File: test/e2e/tests/change-language.spec.js (hunk 1 above)
Comment (hjetpoluru) [reply test/e2e/tests/change-language.spec.js]: I made the code changes and marking resolved.

**Review by NidhiKJha — DISMISSED** [review]

**Review by danjm — DISMISSED** [review]

File: test/scenarios/17. settings/general/change-language.csv [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Verify that the application handles different local times appropriately
+
+Step,Test Steps,Test Data,Expected Result,Notes
+1,Open the extension.,,The Welcome Back screen is shown,
+2,Proceed to Unlock the wallet.,password (8 characters min),,
+3,Click on the right corner setting(three dots).,,General settings page appears,
+4,Check the default language selected in the drop down For example :- local time in the Mac machine is set to 3:30 pm EST,,Default selected language should be English,
```
Comment (DDDDDanica) [inline test/scenarios/17. settings/general/change-language.csv:7]: Question: is 3:30 pm persistent for test case? Maybe i'm missing something but i didn't see any configuration

Comment (hjetpoluru) [reply test/scenarios/17. settings/general/change-language.csv:7]: Its a manual test case and an example for the local time could be anything and the drop down option should reflect the region - language. 
Also, could change the local time manually and see the drop down language change happens.

Comment (DDDDDanica) [reply test/scenarios/17. settings/general/change-language.csv:7]: clear, thank you !

**Review by DDDDDanica — DISMISSED** [review]

**Comment by HowardBraham** [comment]
@hjetpoluru Several of your new tests are failing, especially
"Settings - general tab, validate the change language functionality: User selects "मानक हिन्दी" language and verify that tooltips are updated with the selected language change"

**Comment by HowardBraham** [comment]
It was having trouble clicking the `await driver.clickElement(selectors.appHeaderLogo);` statements, so I changed them to `await driver.navigate();`, which accomplishes the same thing.  I also added a `@no-mmi` tag to one test.  Now it's passing CI.

**Review by HowardBraham — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM! Please update also the test coverage excel file to track when the e2e test and manual scenarios documentation for this feature are added when this PR is merged.

**Pull request merged**